app/models/service_retire_task.rb

```diff
@@ -42,6 +41,7 @@ def create_retire_subtasks(parent_service)
       # Initial Options[:dialog] to an empty hash so we do not pass down dialog values to child services tasks
       nh['options'][:dialog] = {}
       new_task = create_task(svc_rsc, parent_service, nh)
```

Comment:
@d-m-u We need to pass the task(as parent_task) into the create_task method.

Then, we should set the parent_task_id to the parent_task.id here:
https://github.com/ManageIQ/manageiq/blob/352693b3de1c0cee4945bc2b866101ed8c7073d2/app/models/service_retire_task.rb#L58 

And, we should set the parent_task.miq_request_tasks accordingly:
https://github.com/ManageIQ/manageiq/blob/352693b3de1c0cee4945bc2b866101ed8c7073d2/app/models/service_retire_task.rb#L60    
parent_task.miq_request_tasks << new_task